decoder.go

```diff
@@ -34,28 +34,31 @@ func NewDecoder(r io.Reader) *Decoder {
 // Graceful disconnects (between events) are indicated by an io.EOF error.
 // Any error occuring mid-event is considered non-graceful and will
 // show up as some other error (most likely io.ErrUnexpectedEOF).
-func (dec *Decoder) Decode() (Event, error) {
+// Decoding will return either a successfully parsed Event, a comment, or
+// an error.
+func (dec *Decoder) Decode() (Event, *string, error) {
 
 	// peek ahead before we start a new event so we can return EOFs
 	_, err := dec.Peek(1)
 	if err == io.ErrUnexpectedEOF {
 		err = io.EOF
 	}
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	pub := new(publication)
 	for {
 		line, err := dec.ReadString('\n')
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		if line == "\n" {
 			break
 		}
 		line = strings.TrimSuffix(line, "\n")
 		if strings.HasPrefix(line, ":") {
-			continue
+			comment := line[1:]
+			return nil, &comment, nil
```

Comment:
It would probably be good to add a test in `codec_test` for comments in the stream.